api/konnect/v1alpha1/konnect_gateway_controlplane_types.go

```diff
@@ -88,6 +93,18 @@ func (c *KonnectGatewayControlPlane) GetKonnectAPIAuthConfigurationRef() Konnect
 	return c.Spec.KonnectConfiguration.APIAuthConfigurationRef
 }
 
+// SetMemberStatus sets conditions of a member of the control plane group in status.
+// REVIEW: is it better to replace it with `SetMemberCondition` to set a single condition of a member?
```

Comment:
How about not adding another field and having it granular per member, but only populating a `MembersReferencesResolved` condition in the CP's conditions? That would be `True` if all are valid, and `False` if any are invalid (with a message indicating which ones).

When creating https://github.com/Kong/gateway-operator/issues/797 I didn't take that into account, but I think that might be easier to implement and cleaner.

Reply:
👍 This would already be a nice improvement and as you've mentioned: will be easier, less error prone and more efficient to implement.

WDYT @randmonkey ?

Reply:
It sounds good to add a specific condition type in the CP representing the CP group. For further details, we can add fields `map[string]MemeberStatus` for status of each member. The detail part could be in the next release. How about this?